test/e2e/mcd_test.go

```diff
@@ -40,3 +46,103 @@ func TestMCDToken(t *testing.T) {
 		}
 	}
 }
+
+func mcLabelForWorkers() map[string]string {
+	mcLabels := make(map[string]string)
+	mcLabels["machineconfiguration.openshift.io/role"] = "worker"
+	return mcLabels
+}
+
+func createMCFile(path, content string, mode int) ignv2_2types.File {
+	return ignv2_2types.File{
+		FileEmbedded1: ignv2_2types.FileEmbedded1{
+			Contents: ignv2_2types.FileContents{
+				Source: content,
+			},
+			Mode: &mode,
+		},
+		Node: ignv2_2types.Node{
+			Filesystem: "root",
+			Path:       path,
+		},
+	}
+}
+
+func TestMCDeployed(t *testing.T) {
+	cb, err := common.NewClientBuilder("")
+	if err != nil {
+		t.Errorf("%#v", err)
+	}
+	mcClient := cb.MachineConfigClientOrDie("mc-file-add")
+	k := cb.KubeClientOrDie("mc-file-add")
+
+	// create a dummy MC
+	mcName := fmt.Sprintf("00-0add-a-file-%s", uuid.NewUUID())
+	mcadd := &mcv1.MachineConfig{}
+	mcadd.ObjectMeta = metav1.ObjectMeta{
+		Name:   mcName,
+		Labels: mcLabelForWorkers(),
+	}
+	mcadd.Spec = mcv1.MachineConfigSpec{
+		Config: ignv2_2types.Config{
+			Ignition: ignv2_2types.Ignition{
+				Version: "2.2.0",
+			},
+			Storage: ignv2_2types.Storage{
+				Files: []ignv2_2types.File{
+					createMCFile("/etc/mytestconf", "data:,test", 420),
+				},
+			},
+		},
+	}
+
+	// create the dummy MC now
+	_, err = mcClient.MachineconfigurationV1().MachineConfigs().Create(mcadd)
+	if err != nil {
+		t.Errorf("failed to create machine config %v", err)
+	}
+
+	// grab the latest worker- MC
+	var newMCName string
+	err = wait.Poll(2*time.Second, 5*time.Minute, func() (bool, error) {
+		mcp, err := mcClient.MachineconfigurationV1().MachineConfigPools().Get("worker", metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+		for _, mc := range mcp.Status.Configuration.Source {
+			if mc.Name == mcName {
+				newMCName = mcp.Status.Configuration.Name
+				return true, nil
+			}
+		}
+		return false, nil
+	})
+
+	listOptions := metav1.ListOptions{
+		LabelSelector: labels.SelectorFromSet(labels.Set{"k8s-app": "machine-config-daemon"}).String(),
+	}
+
+	err = wait.Poll(3*time.Second, 5*time.Minute, func() (bool, error) {
```

Comment:
Minor: hmm, so this is going over every log of every pod? Could we make this smarter by only going over pods on worker nodes?

Reply:
I could do that yeah

Reply:
It's also only the MCD pods right?

Reply:
yeah, should be https://github.com/openshift/machine-config-operator/pull/387#discussion-diff-254722357R122

Reply:
Yeah, true. So every log of every MCD pod, rather than just worker MCD pods. Guess if it's not hard to select for only worker nodes, it'd be a nice addition, otherwise meh!

Reply:
I'm checking that, but MCDs aren't referencing worker or master explicitly so it's gonna be a great dance to get worker nodes and then match on names, what do you think?